nav2-usability/scenario-code/skipping-stacks/router.dart

```diff
@@ -0,0 +1,503 @@
+// Copyright 2021, the Flutter project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+
+final List<Book> books = [
+  Book('Stranger in a Strange Land', Author('Robert A. Heinlein')),
+  Book('Foundation', Author('Isaac Asimov')),
+  Book('Fahrenheit 451', Author('Ray Bradbury')),
+];
+
+List<Author> get authors => [
+  ...books.map((book) => book.author),
+];
+
+void main() {
+  runApp(BooksApp());
+}
+
+class Book {
+  final String title;
+  final Author author;
+
+  Book(this.title, this.author);
+}
+
+class Author {
+  String name;
+
+  Author(this.name);
+}
+
+class AppState extends ChangeNotifier {
+  bool _showAuthorsPage = false;
+  int? _selectedBookId;
+  int? _selectedAuthorId;
+
+  int? get selectedBookId => _selectedBookId;
+
+  int? get selectedAuthorId => _selectedAuthorId;
+
+  Book? get selectedBook {
+    var idx = _selectedBookId;
+    if (idx == null) {
+      return null;
+    }
+    return books[idx];
+  }
+
+  set selectedBookId(int? id) {
+    if (id == null || id < 0 || id > books.length - 1) {
+      return;
+    }
+
+    _showAuthorsPage = false;
+    _selectedBookId = id;
+    _selectedAuthorId = null;
+    notifyListeners();
+  }
+
+  set selectedBook(Book? book) {
+    if (book == null) {
+      selectedBookId = null;
+      return;
+    }
+    selectedBookId = books.indexOf(book);
+  }
+
+  Author? get selectedAuthor {
+    var idx = _selectedAuthorId;
+    if (idx == null) {
+      return null;
+    }
+    return authors[idx];
+  }
+
+  set selectedAuthorId(int? id) {
+    if (id == null || id < 0 || id > books.length - 1) {
+      return;
+    }
+
+    _selectedAuthorId = id;
+    _showAuthorsPage = false;
+    _selectedBookId = null;
+    notifyListeners();
+  }
+
+  set selectedAuthor(Author? author) {
+    if (author == null) {
+      selectedAuthorId = null;
+      return;
+    }
+    selectedAuthorId = authors.indexOf(author);
+  }
+
+  set showAuthorsPage(bool show) {
```

Comment:
`showAuthorsPage` sounds like a callback,  Maybe `inAuthorsPage`?

Reply:
changed to `isViewingAuthorsPage`